File: js/test/util.spec.ts
import { suite, test } from "mocha-typescript";
import { expect } from "chai";
import { dbl, wipe } from "../src/internal/util";

import { DblExample } from "./support/test_vectors";

@suite class DblSpec {
  static vectors: DblExample[];

  static async before() {
    this.vectors = await DblExample.loadAll();
  }

  @test async "passes the AES-CMAC test vectors"() {
    let result = new Uint8Array(16);

    for (let v of DblSpec.vectors) {
      dbl(v.input, result);
      expect(result).to.eql(v.output);
    }
  }
}

@suite class WipeSpec {
  @test "should wipe bytes"() {
    const a = new Uint8Array([1, 2, 3, 4]);
    wipe(a);
    expect(a).to.eql(new Uint8Array(4));
  }
}

File: ruby/spec/miscreant/internals/util_spec.rb
# frozen_string_literal: true

RSpec.describe Miscreant::Internals::Util do
  describe "dbl" do
    let(:test_vectors) { described_class::DblExample.load_file }

    it "passes all test vectors" do
      test_vectors.each do |ex|
        result = described_class.dbl(ex.input)
        expect(result).to eq(ex.output)
      end
    end
  end
end
